packages/cli/src/commonlib/codeFlowLogin.ts

```diff
@@ -378,26 +377,26 @@ export class CodeFlowLogin {
   }
 }
 
-function sendFile(
+async function sendFile(
   res: http.ServerResponse,
   filepath: string,
   contentType: string,
   accountName: string
-) {
-  fs.readFile(filepath, (err, body) => {
-    if (err) {
-      CliCodeLogInstance.necessaryLog(LogLevel.Error, err.message);
-    } else {
-      let data = body.toString();
-      data = data.replace(/\${accountName}/g, accountName == "azure" ? "Azure" : "M365");
-      body = Buffer.from(data, UTF8);
-      res.writeHead(200, {
-        "Content-Length": body.length,
-        "Content-Type": contentType,
-      });
-      res.end(body);
-    }
-  });
+): Promise<void> {
+  try {
+    let body = await fs.readFile(filepath);
+    let data = body.toString();
+    data = data.replace(/\${accountName}/g, accountName == "azure" ? "Azure" : "M365");
+    body = Buffer.from(data, UTF8);
+    res.writeHead(200, {
+      "Content-Length": body.length,
+      "Content-Type": contentType,
+    });
+    res.end(body);
```

Comment:
Is `res.end` sync/async/callback?

Reply:
```end(chunk: any, cb?: () => void): void;```  is a sync method. And we don't use callback parameter.

Reply:
So the return for `end()` does not close the response, but `cb` means the response is ended?